src/util/fetchUtils.js

```diff
@@ -28,36 +28,33 @@ export const parseQueryResponse = queries => ({ responses, error, message }) =>
 };
 
 /**
- * Wrapper around `fetch` to send an array of queries to the server. It ensures
  * that the request will have the required OAuth and CSRF credentials and constructs
  * the `fetch` call arguments based on the request method. It also records the
  * CSRF header value in a cookie for use as a CSRF header in future fetches.
  *
- * **IMPORTANT**: This function should _only_ be called from the browser. The
- * server should never need to call itself over HTTP
- *
  * @param {String} apiUrl the general-purpose endpoint for API calls to the
  *   application server
  * @param {Object} options {
  *     method: "get", "post", "delete", or "patch",
  *   }
- * @return {Promise} resolves with a `{queries, responses}` object
+ * @param {Array} queries the queries to send - must all use the same `method`
+ * @param {Object} meta additional characteristics of the request, e.g. logout,
+ *   click tracking data
+ * @return {Object} { url, config } arguments for a fetch call
  */
-export const fetchQueries = (apiUrl, options) => (queries, meta) => {
-	if (
-		typeof window === 'undefined' &&  // not in browser
-		typeof test === 'undefined'  // not in testing env
-	) {
-		throw new Error('fetchQueries was called on server - cannot continue');
-	}
-	options.method = options.method || 'GET';
+export const getFetchArgs = (apiUrl, options, queries, meta) => {
 	const {
-		method,
 		headers={},
 	} = options;
 
-	const isPost = method.toLowerCase() === 'post';
-	const isDelete = method.toLowerCase() === 'delete';
+	const method = ((queries[0].meta || {}).method || '').toLowerCase() ||  // allow query to set method
+		options.method.toLowerCase() ||  // fallback to options
+		'get';  // fallback to 'get'
+
+	const hasBody = method === 'post' ||
```

Comment:
Unfortunately, we will also have to support `DELETE` requests with bodies. For example: https://www.meetup.com/meetup_api/docs/:urlname/events/:id/#delete

Maybe we can add that support in the same PR?